crates/wast/src/core/types.rs

```diff
@@ -505,6 +505,17 @@ impl<'a> Parse<'a> for StorageType<'a> {
     }
 }
 
+impl From<StorageType<'_>> for wasm_encoder::StorageType {
```

Comment:
Mind moving these new `From` impls to `core/binary.rs`? That matches the style of `component/binary.rs` and as we transition to wasm-encoder over time for core bits I'm thinking that more things will pop up there